bundles/admin/admin-hierarchical-layerlist/views/adminLayerSettingsView.js

```diff
@@ -0,0 +1,1545 @@
+                'click .add-dataprovider-button': 'addDataprovider',
+                'click .import-wfs-style-button': 'importSldStyle',
+                'click .save-wfs-style-button': 'saveSldStyle',
+                'click .cancel-wfs-style-button': 'cancelSldStyle',
+                'click .icon-close': 'clearInput',
+                'change .admin-layer-type': 'createLayerSelect',
+                'click .admin-add-group-ok': 'saveCollectionLayer',
+                'click .admin-add-group-cancel': 'hideLayerSettings',
+                'click .admin-remove-group': 'removeLayerCollection',
+                'click .add-layer-record.capabilities li': 'handleCapabilitiesSelection',
+                'change .admin-interface-version': 'handleInterfaceVersionChange',
+                'change .admin-sld-styles': 'handleSldStylesChange',
+                'change .admin-layer-legendUrl': 'handleLayerLegendUrlChange',
+                'click .layer-capabilities.icon-info': 'showCapabilitiesPopup',
+                "click .admin-add-sublayer": "toggleSubLayerSettings",
+                "click .select-maplayer-groups-button": "selectMaplayerGroups"
+            },
+            /*******************************************************************************************************************************
+            /* PRIVATE METHODS
+            *******************************************************************************************************************************/
+            /**
+             * Save dataprovider
+             * @method  _saveDataprovider
+             * @param   {Object}   data  data fo saving
+             * @param   {Oskari.userinterface.component.Popup}   popup group adding/editing popup
+             * @private
+             */
+            _saveDataprovider: function(data, popup) {
+                var me = this;
+
+                jQuery.ajax({
+                    type: 'PUT',
+                    url: me.instance.sandbox.getAjaxUrl('SaveOrganization'),
+                    data: data,
+                    error: function() {
+                        var errorDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+                        errorDialog.show(me.instance.locale('errors.dataproviderSave.title'), me.instance.locale('errors.dataproviderSave.message'));
+                        errorDialog.fadeout();
+                    },
+                    success: function(response) {
+                        popup.close();
+                        var successDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+                        successDialog.show(me.instance.locale('succeeses.dataproviderSave.title'), me.instance.locale('succeeses.dataproviderSave.message'));
+                        successDialog.fadeout();
+
+                        jQuery('#select-dataprovider').append('<option value="' + response.id + '">' + Oskari.getLocalized(response.name) + '</option>');
```

Comment:
Do you remove the previous option while editing somewhere? Or does editing leave the old option and add the new option?